src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py

```diff
@@ -4182,108 +4183,54 @@ def test_sql_managed_instance_mgmt(self):
-        proxy_override = "Proxy"
-
-        test_umi = '/subscriptions/e64f3e8e-ab91-4a65-8cdd-5cd2f47d00b4/resourceGroups/viparek/providers/Microsoft.ManagedIdentity/userAssignedIdentities/testumi'
-        umi_list = '/subscriptions/e64f3e8e-ab91-4a65-8cdd-5cd2f47d00b4/resourceGroups/viparek/providers/Microsoft.ManagedIdentity/userAssignedIdentities/testumi'
-        
-        identity_type = ResourceIdType.system_assigned_user_assigned.value
-        user = admin_login
-
-        # test create another sql managed instance, with identity this time
-        self.cmd('sql mi create -g {} -n {} -l {} -i '
-                 '--admin-user {} --admin-password {} --subnet {} --license-type {} --capacity {} --storage {} --edition {} --family {} --collation {} --proxy-override {} --public-data-endpoint-enabled ' 
-                  '--user-assigned-identity-id {} --identity-type {} --pid {}'
-                 .format(resource_group_1, managed_instance_name, loc, user, admin_passwords[0], subnet, license_type, v_cores, storage_size_in_gb, edition,
-                  families[0], collation, proxy_override, umi_list, identity_type, test_umi),
-                 checks=[
-                     JMESPathCheck('name', managed_instance_name),
-                     JMESPathCheck('resourceGroup', resource_group_1),
-                     JMESPathCheck('administratorLogin', user),
-                     JMESPathCheck('vCores', v_cores),
-                     JMESPathCheck('storageSizeInGb', storage_size_in_gb),
-                     JMESPathCheck('licenseType', license_type),
-                     JMESPathCheck('sku.tier', edition),
-                     JMESPathCheck('sku.family', families[0]),
-                     JMESPathCheck('sku.capacity', v_cores),
-                     JMESPathCheck('identity.type', 'SystemAssigned, UserAssigned'),
-                     JMESPathCheck('collation', collation),
-                     JMESPathCheck('proxyOverride', proxy_override),
-                     JMESPathCheck('publicDataEndpointEnabled', 'True')])
+    @ManagedInstancePreparer(
+        identity_type=ResourceIdType.system_assigned_user_assigned.value,
+        user_assigned_identity_id=test_umi,
+        pid=test_umi)
+    def test_sql_managed_instance_create_identity_mgmt(self, mi, rg):
+        managed_instance_name = mi
+        resource_group_1 = rg
 
         # test show sql managed instance 2
         self.cmd('sql mi show -g {} -n {}'
                  .format(resource_group_1, managed_instance_name),
                  checks=[
                      JMESPathCheck('name', managed_instance_name),
                      JMESPathCheck('resourceGroup', resource_group_1),
-                     JMESPathCheck('administratorLogin', user)])
-
-        self.cmd('sql mi delete -g {} -n {} --yes'
-                 .format(resource_group_1, managed_instance_name), checks=NoneCheck())
+                     JMESPathCheck('primaryUserAssignedIdentityId', self.test_umi),
```

Comment:
for this test, I also met the same situation as I commented in: https://github.com/Azure/azure-cli/pull/18637#issuecomment-876825197

I passed this test locally. but it failed in pipeline due to empty guide:
https://dev.azure.com/azure-sdk/public/_build/results?buildId=999285&view=logs&j=15e5eb7c-cf66-5b91-00a2-db9b11b5e0f9&t=845fa2ee-a602-59f5-4e4c-07ee6d0283b9&l=5679

E               azure.cli.testsdk.exceptions.JMESPathCheckAssertionError: Query 'primaryUserAssignedIdentityId' doesn't yield expected value '/subscriptions/e64f3e8e-ab91-4a65-8cdd-5cd2f47d00b4/resourcegroups/viparek/providers/Microsoft.ManagedIdentity/userAssignedIdentities/testumi', instead the actual value is '/subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/viparek/providers/Microsoft.ManagedIdentity/userAssignedIdentities/testumi'. Data: 


do we have any way to resolve this issue?

Reply:
There are two ways:
1. do not check full resource id for user assigned identity. Name is enough.
2. Replace subscription information with 00000000-0000-0000-0000-000000000000 in playback mode.

Reply:
addressed with if condition of is_live/is_recording